chains/substrate/connection/connection.go

```diff
@@ -0,0 +1,91 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: BUSL-1.1
+
+package connection
+
+import (
+	"sync"
+
+	"github.com/ChainSafe/sygma-relayer/chains/substrate/events"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/client"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/rpc/chain"
+
+	"github.com/centrifuge/go-substrate-rpc-client/v4/rpc"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+type Connection struct {
+	chain.Chain
+	*rpc.RPC
+	meta        types.Metadata // Latest chain metadata
+	metaLock    sync.RWMutex   // Lock metadata for updates, allows concurrent reads
+	genesisHash types.Hash     // Chain genesis hash
+	name        string         // Chain name
+}
+
+func NewSubstrateConnection(url string, name string) (*Connection, error) {
```

Comment:
Why do we need the name?

Reply:
Might be needed for some logging stuff... I can remove it for now, and add it later if needed?